public/app/features/dashboard/containers/DashboardPage.tsx

```diff
@@ -273,7 +273,7 @@ export class UnthemedDashboardPage extends PureComponent<Props, State> {
     else if (state.viewPanel && !urlViewPanelId) {
       // This mutable state feels wrong to have in getDerivedStateFromProps
       // Should move this state out of dashboard in the future
-      dashboard.exitViewPanel(state.viewPanel);
+      dashboard.exitViewPanel(dashboard.panelInView ?? state.viewPanel);
```

Comment:
I really don't understand how this is fixing the issue of the panel staying huge after leaving the View Panel mode if you have refreshed it at least once.. since the `dashboard.panelInView` is the first one to become `undefined`, so it shouldn't tip the scales of what's happening here. 

But it does, somehow 😅. I love this, @kaydelaney, I would like it if you explain to me why it works.